scalyr_agent/builtin_monitors/kubernetes_monitor.py

```diff
@@ -2262,13 +2262,23 @@ def _get_containers(
                     # NOTE: CRIEnumerator doesn't utilize ControlledCacheWarmer so it's important
                     # we use allow_expired=False here otherwise we will always read cached entry
                     # even if it's stale
-                    pod = k8s_cache.pod(
-                        pod_namespace,
-                        pod_name,
-                        current_time,
-                        allow_expired=False,
-                        ignore_k8s_api_exception=True,
-                    )
+
+                    pod = None
+                    try:
+                        pod = k8s_cache.pod(
+                            pod_namespace,
+                            pod_name,
+                            current_time,
+                            allow_expired=False,
+                            ignore_k8s_api_exception=False,
+                        )
+                    except k8s_utils.K8sApiException as e:
```

Comment:
It might be good to use the http status code from the K8sApiException to help decide what to do

For example looking at the notes from DTIN-3156, a 404 means the pod doesn't exist so it should always be excluded

Reply:
That makes sense.
404 might also mean the pod existed and then was garbage collected before the agent had a chance to process its logs.
But in that case, we wouldn't know if it was annotated to be included anyway.
So the safe action is to exclude it and log a warning.

Reply:
404 => Pod not found, Log Warning, excluded
401 => Unauthorized, Log Error, excluded